src/library/legacylibraryimporter.cpp

```diff
@@ -34,9 +34,16 @@ LegacyLibraryImporter::LegacyLibraryImporter(TrackDAO& trackDao,
 
 /** Upgrade from <= 1.7 library to 1.8 DB format */
 void LegacyLibraryImporter::import() {
-    // TODO(XXX) SETTINGS_PATH may change in new Mixxx Versions. Here we need
-    // the SETTINGS_PATH from Mixxx V <= 1.7
-    QString settingPath17 = QDir::homePath().append("/").append(SETTINGS_PATH);
+    //Here we need the SETTINGS_PATH from Mixxx V <= 1.7
+#ifdef __LINUX__
+    QString settingPath17 = QDir::homePath().append("/").append(".mixxx/");
+#elif __WINDOWS__
+    QString settingPath17 = QDir::homePath().append("/").append("Local Settings/Application Data/Mixxx/");
+#elif __APPLE__
+    QString settingPath17 = QDir::homePath().append("/").append(".mixxx/");
```

Comment:
No,  on osx, this has changed from .mixxx to Library... in 1.9
We need 1.7 value here as stated in the above comment. 

This compatibility code didn't work on OSX since 1.9